s3/src/main/scala/monix/connect/s3/S3.scala

```diff
@@ -234,41 +581,51 @@ object S3 {
     *
     * @see https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/model/DeleteObjectRequest.html
-    * @param request the AWS delete object request of type [[DeleteObjectRequest]]
-    * @param s3AsyncClient implicit instance of a [[S3AsyncClient]].
+    * @param request       the AWS delete object request of type [[DeleteObjectRequest]]
     * @return a [[Task]] with the delete object response [[DeleteObjectResponse]] .
     */
-  def deleteObject(request: DeleteObjectRequest)(implicit s3AsyncClient: S3AsyncClient): Task[DeleteObjectResponse] =
-    Task.from(s3AsyncClient.deleteObject(request))
+  def deleteObject(request: DeleteObjectRequest): Task[DeleteObjectResponse] =
+    Task.from(s3Client.deleteObject(request))
 
   /**
     * Check whether the specified bucket exists or not.
     *
-    * @param bucket the bucket name to check its existence
-    * @param s3AsyncClient implicit instance of a [[S3AsyncClient]].
+    * @param bucket        the bucket name to check its existence
     * @return a boolean [[Task]] indicating whether the bucket exists or not.
     */
-  def existsBucket(bucket: String)(implicit s3AsyncClient: S3AsyncClient): Task[Boolean] =
-    S3.listBuckets().existsL(_.name == bucket)
+  def existsBucket(bucket: String): Task[Boolean] =
+    self.listBuckets().existsL(_.name == bucket)
 
   /**
-    * Checks whether the specified objects exists or not.
+    * Checks whether the specified object exists or not.
+    *
+    * ==Example==
     *
-    * @param bucket the bucket name of the object to check its existence.
-    * @param key the key of the object to be deleted.
-    * @param s3AsyncClient implicit instance of a [[S3AsyncClient]].
+    * {{{
+    *   import monix.eval.Task
+    *   import monix.connect.s3.S3
+    *   import cats.effect.Resource
+    *
+    *   val s3Resource: Resource[Task, S3] = S3.create()
+    *
+    *   val bucket = "my-bucket"
+    *   val s3Key = "my-key"
+    *
+    *   val t: Task[Boolean] = s3Resource.use(_.existsObject(bucket, s3Key))
+    * }}}
+    *
+    * @param bucket        the bucket name of the object to check its existence.
+    * @param key           the key of the object to be deleted.
     * @return a boolean [[Task]] indicating whether the object existed or not.
     */
-  def existsObject(bucket: String, key: String)(implicit s3AsyncClient: S3AsyncClient): Task[Boolean] = {
-    Task.defer {
-      Task.from {
-        s3AsyncClient.headObject(S3RequestBuilder.headObjectRequest(bucket, Some(key)))
-      }
-    }.redeemWith(
-      ex =>
-        if (ex.isInstanceOf[NoSuchKeyException]) Task.now(false)
-        else Task.raiseError(ex),
-      _ => Task.now(true))
+  def existsObject(bucket: String, key: String): Task[Boolean] = {
+    Task
+      .from(s3Client.headObject(S3RequestBuilder.headObjectRequest(bucket, Some(key))))
+      .redeemWith(
+        ex =>
+          if (ex.isInstanceOf[NoSuchKeyException]) Task.now(false)
```

Comment:
As a slight optimization, we could define `val falseTask = Task.now(false)` and `trueTask` to reuse the reference